src/mgmt/srv.c

```diff
@@ -59,23 +60,104 @@ static struct daos_rpc_handler mgmt_handlers[] = {
 
 #undef X
 
+static void
+process_killrank_request(Drpc__Call *drpc_req, Proto__DaosResponse *daos_resp)
+{
+	Proto__DaosRank	*pb_rank = NULL;
+
+	proto__daos_response__init(daos_resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_rank = proto__daos_rank__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_rank == NULL) {
+		daos_resp->status = PROTO__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
```

Comment:
I really don't like those very long names ...

Reply:
We shall fix the proto files someday; their package option should not all be "proto".